### ISSUE TITLE
Support x-extensions in spectral rulesets

### ISSUE THREAD

**Issue by vincentklg** [stoplightio/spectral#2394]
**Describe the bug**
When declaring an anchor in the ruleset.yml, spectral exits with `Error #1: undefined`
In the example below the anchor is `valid-head-pattern`, we would like to avoid a redundant regex declaration.

**To Reproduce**
1. Given this ruleset file:
```yaml
extends: ["spectral:oas"]

x-vars:
  - &valid-header-pattern ^(Accept|Accept-CH|Accept-CH-Lifetime|Accept-Charset|Accept-Encoding)$

rules:

  invalid-response-header:
      description: "Use of headers not listed in NLB API guide is not allowed."
      severity: error
      given: $..responses..headers
      resolved: false
      then:
        field: "@key"
        function: pattern
        functionOptions:
          match: *valid-header-pattern

```

3. Run this CLI command `spectral lint openapi.yml --ruleset ruleset.yml`
Consider `openai.yml` a valid oas file.

4. See error:
```
Error running Spectral!
Use --verbose flag to print the error stack.
Error #1: undefined
```

**Expected behavior**
I would expect that the alias inside the rule is resolved against the corresponding anchor at the beginning of the file.

**Environment (remove any that are not applicable):**
 - Spectral CLI version: [6.6.0]
 - OS: [e.g. macOS 10.15.7] I know 🙄

**Comment by P0lip** [comment]
It's not an issue with YAML anchors per se, but with ruleset having additional properties.
As things stand, we don't allow any x- extensions to be provided, so `x-vars` triggers a validation error.

if you do the following

```yaml
extends: ["spectral:oas"]



rules:

  invalid-response-header:
    description: "Use of headers not listed in NLB API guide is not allowed."
    severity: error
    given: $..responses..headers
    resolved: false
    then:
      - field: "@key"
        function: pattern
        functionOptions:
          match: &valid-header-pattern "^(Accept|Accept-CH|Accept-CH-Lifetime|Accept-Charset|Accept-Encoding)$"
      - field: "@key"
        function: pattern
        functionOptions:
          match: *valid-header-pattern
```

it works just fine.

**Comment by vincentklg** [comment]
Hi @P0lip, tanks for your response, works as described.
I didn't think of declaring an anchor inside a rule.
On my end the issue is resolved, but I am not going to close it since you added the enhancement tag.
But feel free to close it.
